src/jit/lower.h

```diff
@@ -143,8 +143,28 @@ class Lowering : public Phase
     void TreeNodeInfoInit(GenTreePtr* tree, GenTree* parent);
 #if defined(_TARGET_XARCH_)
     void TreeNodeInfoInitSimple(GenTree* tree);
-    void SetRegOptionalForBinOp(GenTree* tree);
-    void TryToSetRegOptional(GenTree* operand);
+
+    //----------------------------------------------------------------------
+    // SetRegOptional - sets a bit to indicate to LSRA that register
+    // for a given tree node is optional for codegen purpose.  If no
+    // register is allocated to such a tree node, its parent node treats
+    // it as a contained memory operand during codegen.
+    //
+    // Arguments:
+    //    tree    -   GenTree node
+    //
+    // Returns
+    //    None
+    //
+    // Note: By modifying this routine we can control which of the
+    // gentree nodes get marked as reg-optional.
+    void SetRegOptional(GenTree* tree)
```

Comment:
What does this note mean? In what way would this routine be modified, and for what purpose?